File: backend/Models/Comment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TaskManagementApi.Models
{
    public class Comment
    {
        [Key]
        public required int Id { get; set; }

        // We only need the TaskId foreign key
        public required int TaskId { get; set; }

        public required string Content { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? LastUpdated { get; set; }

        public required string CreatorId { get; set; }

        // For reply functionality
        public int? ParentCommentId { get; set; }
        
        // We can keep the Replies collection for convenience
        [ForeignKey("ParentCommentId")]
        public virtual ICollection<Comment>? Replies { get; set; } = new List<Comment>();
    }
}

File: front/src/types/comment.ts
export interface Comment {
  id: number;
  taskId: number;
  content: string;
  createdBy: string;
  createdAt: string;
  lastUpdated?: string;
  parentCommentId?: number;
  replies?: Comment[];
}

export interface CommentFormData {
  content: string;
  parentCommentId?: number;
}

export interface AddCommentInput {
  taskId: number;
  content: string;
  createdBy: string;
  parentCommentId?: number;
}

export interface UpdateCommentInput {
  id: number;
  content: string;
}
